docs/DesignDocs/Attestation_API_Proposal.md

```diff
@@ -592,62 +593,106 @@ plugin libraries, composed of:
+* `OE_FORMAT_UUID_SGX_LOCAL_ATTESTATION`:
+  * The input evidence is generated by `oe_get_evidence()` for format
+  `OE_FORMAT_UUID_SGX_LOCAL_ATTESTATION`, with the
+  `OE_EVIDENCE_FLAGS_EMBED_FORMAT_ID` bit cleared in its `flags` parameter.
+* `OE_FORMAT_UUID_SGX_ECDSA_P256`:
+  * The input evidence is generated by `oe_get_evidence()` for format
+  `OE_FORMAT_UUID_SGX_ECDSA_P256`, with the
+  `OE_EVIDENCE_FLAGS_EMBED_FORMAT_ID` bit cleared in its `flags` parameter.
+* `OE_FORMAT_UUID_SGX_REPORT`:
+  * The input evidence is an OE report generated by the legacy API function
+  `oe_get_report()`.
+* `OE_FORMAT_UUID_SGX_QUOTE`:
+  * The input evidence is a raw SGX ECDSA quote generated by the
+  Intel SGX SDK DCAP library.
+
+The table below shows the structure of the evidence data for all the supported
+SGX format IDs, as generated by an attester plugin or verified by a verifier
+plugin.
 
 | Format ID | Evidence structure  |
 | -- | - |
-| `OE_FORMAT_UUID_SGX_LOCAL_ATTESTATION` | `oe_report_header \|\| SGX_report(hash) \|\| custom_claims` |
-| `OE_FORMAT_UUID_SGX_ECDSA_P256` | `oe_report_header \|\| SGX_ECDSA_quote(hash) \|\| custom_claims` |
-| `OE_FORMAT_UUID_SGX_ECDSA_P256_REPORT` | `oe_report_header \|\| SGX_ECDSA_quote(custom_claims)` |
-| `OE_FORMAT_UUID_SGX_ECDSA_P256_QUOTE` | `SGX_ECDSA_quote(custom_claims)` |
-| `OE_FORMAT_UUID_SGX_EPID_LINKABLE` | `SGX_EPID_linkable_quote(custom_claims)` |
-| `OE_FORMAT_UUID_SGX_EPID_UNLINKABLE` | `SGX_EPID_unlinkable_quote(custom_claims)` |
+| `OE_FORMAT_UUID_SGX_LOCAL_ATTESTATION` | `[ oe_attestation_header ] \|\| SGX_report(hash) \|\| custom_claims` |
+| `OE_FORMAT_UUID_SGX_ECDSA_P256` | `[ oe_attestation_header ] \|\| SGX_ECDSA_quote(hash) \|\| custom_claims` |
+| `OE_FORMAT_UUID_SGX_EPID_LINKABLE` | `[ oe_attestation_header ] \|\| SGX_EPID_linkable_quote(custom_claims)` |
+| `OE_FORMAT_UUID_SGX_EPID_UNLINKABLE` | `[ oe_attestation_header ] \|\| SGX_EPID_unlinkable_quote(custom_claims)` |
+| `OE_FORMAT_UUID_SGX_REPORT` | `oe_report_header \|\| OE_report_body(custom_claims)` |
+| `OE_FORMAT_UUID_SGX_QUOTE` | `SGX_quote(custom_claims)` |
 
 In the above table:
+* The optional header `oe_attestation_header` is a structure of type
+`oe_attestation_header_t`.
+* For every format supported by `oe_get_evidence()`, the evidence
+will be prefixed with an `oe_attestation_header` when the
+`OE_EVIDENCE_FLAGS_EMBED_FORMAT_ID` bit in its `flags` parameter is set.
 * `oe_report_header` is the OE report header of type `oe_report_header_t`.
 * `hash` is the SHA256 hash of the custom claims held in a flat buffer
 `custom_claims`.
-* An SGX report (`SGX_report()`) or quote (`SGX_*_quote()`) contains a
+* An SGX report (`SGX_report()`) or quote (`SGX_*quote()`) embeds a
 flat buffer of 64 bytes for its SGX report data field. Depending on the format,
 this field holds either the `hash` of the custom claims, or the custom claims
 directly.
+* Note: for all evidence generated by API `oe_get_evidence()`, the
```

Comment:
This will not happen in release 0.10.1. The verification support of the new format should be implemented first before changing the evidence generation.

Reply:
> This will not happen in release 0.10.1. The verification support of the new format should be implemented first before changing the evidence generation.

Do you mean we should not implement it in 0.10.1? After approval of this PR, there will be another code PR that implements all the proposed changes, targeted for 0.10.1.

Reply:
If we change the evidence structure generated by oe_get_evidence(), apps that built with older OE SDK won't be able to call oe_verify_evidence() to verify evidence. I'd suggest we start by allowing oe_verify_evidence() to verify new evidence structure. oe_get_evidence() will follow in the next release.

Reply:
The attestation plugin API is still experimental, that should allow us more freedom to change the definition without having to maintain backward compatibility, especially when oe_attestation_header_t is not exposed as a public structure.

So I still think we can change both ends in a single step. Apps will need to be rebuilt when the new SDK release is used.

In any case, to remove oe_report_header, the version number `OE_ATTESTATION_HEADER_VERSION` will need to be bumped to 3. This allows `oe_verify_evidence()` to detect out-dated evidence and raise error. Or if backward compatibility is still desirable, `oe_verify_evidence()` can keep the logic for version 2 headers.

Reply:
Yes, the version number will need to be bumped to 3. I still prefer 2-phase plan but am open to your plan. Let's discuss in the meeting. There are only a few days left before cut-off time for release 0.10.1. Is your code PR almost ready?